models/error.go

```diff
@@ -76,6 +107,42 @@ func ErrUnmarshal(err error, obj string) error {
 	return errors.New(ErrUnmarshalCode, errors.Alert, []string{"Unable to unmarshal the object", obj}, []string{err.Error()}, []string{}, []string{})
 }
 
+func ErrEncoding(err error, obj string) error {
+	return errors.New(ErrEncodingCode, errors.Alert, []string{"Error encoding the object", obj}, []string{err.Error()}, []string{"Unable to encode the : ", obj}, []string{})
+}
+
+func ErrFetch(err error, obj string, statusCode int) error {
+	return errors.New(ErrFetchCode, errors.Alert, []string{"Unable to fetch data from the Provider", obj}, []string{"Status Code: " + fmt.Sprint(statusCode), err.Error()}, []string{}, []string{})
+}
+
+func ErrPost(err error, obj string, statusCode int) error {
+	return errors.New(ErrPostCode, errors.Alert, []string{"Unable to post data to the Provider", obj}, []string{"Status Code: " + fmt.Sprint(statusCode), err.Error()}, []string{}, []string{})
+}
+
+func ErrDelete(err error, obj string, statusCode int) error {
+	return errors.New(ErrDeleteCode, errors.Alert, []string{"Unable to delete data from the Provider", obj}, []string{"Status Code: " + fmt.Sprint(statusCode), err.Error()}, []string{}, []string{})
+}
+
+func ErrDecodeBase64(err error, obj string) error {
+	return errors.New(ErrDecodeBase64Code, errors.Alert, []string{"Error occurred while decoding base65 string", obj}, []string{err.Error()}, []string{}, []string{})
+}
+
+func ErrMarshalPKIX(err error) error {
```

Comment:
Everything looks great, but just one question we already have ErrMarshal() 
Was there a specific reason to create this function ?

Reply:
@Shreyas220 great catch. @dhruv0000 We will want to reuse the Marshal function.

Reply:
yes. I did thought of it, but this error is specifically for converting a public key to PKIX (X.509) format and not JSON, so I thought it deserved its own error type.

Reply:
If the statement and everything is the same, then irrespective of the use case we could re-use.